base/uk.ac.stfc.isis.ibex.instrument/src/uk/ac/stfc/isis/ibex/instrument/Instrument.java

```diff
@@ -236,7 +237,15 @@ public void setInstrumentForAllPlugins(InstrumentInfo selectedInstrument) {
 
         instrumentName.setValue(selectedInstrument.name());
 
-        updateExtendingPlugins(selectedInstrument);
+        final InstrumentInfo finalSelectedInstrument = selectedInstrument;
+
+        new Thread(new Runnable() {
+            @Override
+            public void run() {
+                updateExtendingPlugins(finalSelectedInstrument);
+            }
+        }).start();
```

Comment:
As below. Why is this running on a separate thread? Obvious from the PR, but assume the person reading it won't want to have to go looking for it.